src/WebJobs.Script/Diagnostics/Extensions/LoggerExtension.cs

```diff
@@ -121,15 +121,15 @@ internal static class LoggerExtension
                 new EventId(325, nameof(ScriptStartUpLoadedExtension)),
                 "Loaded extension '{startupExtensionName}' ({startupExtensionVersion})");
 
-        private static readonly Action<ILogger, Exception> _functionMetadataProviderReadingMetadata =
-            LoggerMessage.Define(LogLevel.Information,
+        private static readonly Action<ILogger, string, Exception> _functionMetadataProviderReadingMetadata =
```

Comment:
Initially the naming for this field and it's corresponding method FunctionMetadataProviderParsingFunctions made sense before worker indexing was added. Now it's being reused in 3 different places: Host file provider, Worker index provider, and Custom providers.

So I think we should rename the property/method to _readingFunctionMetadataFromSource and ReadingFunctionMetadataFromSource respectively